### ISSUE TITLE
MySQLi Class Modifies SQL statement after incorrectly assuming it is creating a table

### ISSUE THREAD

**Issue by eSilverStrike** [Geeklog-Core/geeklog#652]
Was running an import for a plugin and the sql statement below kept failing because "ENGINE=MyISAM DEFAULT CHARSET=utf8" was kept being added at the end of the statement by Geeklogs database class. Looking at the code it looks like in the mysqli.class.php the function dbQuery at line 244 checks to see if it is a create table statement. If it is it then adds the engine info. The problem is that the select statement is not a create table. (it just happens to have those words)
Create Table should exist at the start of the sql statement if not then it is not creating a table and the extra info should not be added.

Should check MySQL.class.php for this bug as well

REPLACE INTO comments (sid,uid,comment,date,title,pid,lft,rht,indent,type,ipaddress,name) VALUES ('50',12279,'This will not form the necessary sql tables what is wrong ?
Sun 17 Sep 09:22:19 2006 - Attempting to install the Journal Plugin
Sun 17 Sep 09:22:19 2006 - Attempting to create table journal for Journal plugin
Sun 17 Sep 09:22:19 2006 - 1067: Invalid default value for jrn_id. SQL in question:

any help gladly recieved<div class=\"comment-sig\">---
One of the Geeklog Core Developers
</div>',now(),'Form wouldn\'t post',15047,6,7,1,'downloads','82.195.106.119','') ENGINE=MyISAM DEFAULT CHARSET=utf8

**Comment by mystralkk** [comment]
What will happen if you change the line 244 from
```PHP
        if (preg_match('/create\s+table\s/i', $sql)) {
```
to
```PHP
        if (preg_match('/^create\s+table\s/i', $sql)) {
```
?

**Comment by eSilverStrike** [comment]
I would think it should work.

**Comment by mystralkk** [comment]
Fixed with changeset dec6623.
